src/box/txn.c

```diff
@@ -1105,8 +1105,15 @@ txn_commit(struct txn *txn)
 			txn_limbo_ack(&txn_limbo, txn_limbo.owner_id,
```

Comment:
> NO_TEST=issue is covered with iproto graceful shutdown

I suggest you add a separate test for this issue. It should be quite easy to write one: you only need one server, set quorum to 2 on it, then create a synchronous transaction in a separate fiber and cancel that fiber. If the change is still seen (without mvcc enabled), the test passes.

> NO_CHANGELOG

But this is a user-visible change: now fiber cancellation doesn't lead to synchronous transaction rollback now. AFAIU this is more consistent with async transactions now. They can't be rolled back by a cancellation after a WAL write is started.

Reply:
Added a test and a changelog:

```md
## bugfix/replication

* Fixed to not rollback a transaction waiting for quorum from replicas
  on transaction fiber cancel (gh-9480).
```

Reply:
Thanks!